### PR TITLE
Avafanasev/qd 9162.242

### PR DESCRIPTION
**Pull request by avafanasiev**
# Pull Request Details

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I have read the [**CONTRIBUTING**](https://github.com/JetBrains/qodana-cli/blob/master/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] My commit messages are styled with [gitmoji](https://gitmoji.dev)
- [ ] My change requires a change to the [documentation](https://jetbrains.com/help/qodana).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: core/license.go [hunk 1]
```diff
@@ -36,6 +36,12 @@ func SetupLicenseAndProjectHash(endpoints *cloud.QdApiEndpoints, token string) {
 				log.Fatal(err)
 			}
 		}
+		if licenseData.OrganisationIdHash != "" {
```
Comment (MekhailS) [inline core/license.go:39]: What happens if in response from server there is no `organizationIdHash`? We already released older on-prem for some customers

File: core/core_test.go [hunk 2]
```diff
@@ -520,33 +520,72 @@ func Test_ideaExitCode(t *testing.T) {
...
+			expectedKey:     "VA5HGQWQH6",
+			expectedHash:    "hash",
+			expectedOrgHash: "org hash",
+		},
+		{
+			name:            "no organizationIdHash",
+			license:         `{"licenseId":"VA5HGQWQH6","licenseKey":"VA5HGQWQH6","expirationDate":"2023-07-31","licensePlan":"EAP_ULTIMATE_PLUS","projectIdHash":"hash"}`,
+			expectedKey:     "VA5HGQWQH6",
+			expectedHash:    "hash",
+			expectedOrgHash: "",
```
Comment (MekhailS) [inline core/core_test.go:542]: Why empty value? there should be no json field at all?

File: core/license.go (hunk 1 above)
Comment (avafanasiev) [reply core/license.go:39]: Where is a test in this commit specificaly for this case.

File: core/core_test.go (hunk 2 above)
Comment (MekhailS) [reply core/core_test.go:542]: Ok, missed that its expected

Comment (avafanasiev) [reply core/core_test.go:542]: That's a go way, string cannot be nil. Logic and test written in a way to not set env variable in a case of empty string.

**Review by MekhailS — APPROVED** [review]

**Comment by MekhailS** [comment]
We are lucky for now thar it's not a separate endpoint

**Pull request merged**